contrib/lua/premake5.lua

```diff
@@ -21,8 +21,8 @@ project "lua-lib"
 		"etc/*.c"
 	}
 
-	configuration "linux or bsd or hurd or aix"
+	filter "system:linux or system:bsd or system:hurd or system:aix"
```

Comment:
It's not obvious to me how a 'term' is defined. Is a scope a term? Am I applying `not` to the whole expression beneath the scope, or the first item listed after the scope, or the scope itself in some way?

What I asked though, was what advantage does supporting that syntax offer? I feel it can only lead to confusion, can you make a case for that expression to exist?